newt/builder/build.go

```diff
@@ -568,6 +571,35 @@ func (b *Builder) Build() error {
 		}
 	}
 
+	var compileCommands []toolchain.CompileCommand
+
+	for _, bpkg := range bpkgs {
+		c := bpkgCompilerMap[bpkg]
+		if c != nil {
+			compileCommands = append(compileCommands,
+				c.GetCompileCommands()...)
+		}
+	}
+
+	projectPath := interfaces.GetProject().Path() + "/"
+	for i := range compileCommands {
+		compileCommands[i].Directory = projectPath
+	}
+
+	cmdBytes, err := json.MarshalIndent(compileCommands, "", "    ")
+	if err != nil {
+		log.Error("Unable to encode compilation commands as JSON")
+		return nil
+	}
+
+	outDir := filepath.Dir(b.AppElfPath())
+	cmdPath := outDir + "/compile_commands.json"
```

Comment:
It might be better to define a function in `builder/paths.go` which returns the compile commands json path (CompileCmdsPath() or similar).  That way, all the paths are defined in the same place.